prow/owners/owners.go

```diff
@@ -0,0 +1,477 @@
+	return result
+}
+
+func loadOwnersFrom(baseDir string, mdYaml bool, aliases RepoAliases, log *logrus.Entry) (*RepoOwners, error) {
+	o := &RepoOwners{
+		RepoAliases:  aliases,
+		baseDir:      baseDir,
+		enableMDYAML: mdYaml,
+		log:          log,
+
+		approvers: make(map[string]sets.String),
+		reviewers: make(map[string]sets.String),
+		labels:    make(map[string]sets.String),
+	}
+
+	return o, filepath.Walk(o.baseDir, o.walkFunc)
+}
+
+// by default, github's api doesn't root the project directory at "/" and instead uses the empty string for the base dir
+// of the project. And the built-in dir function returns "." for empty strings, so for consistency, we use this
+// canonicalize to get the directories of files in a consistent format with NO "/" at the root (a/b/c/ -> a/b/c)
+func canonicalize(path string) string {
+	if path == "." {
+		return baseDirConvention
+	}
+	return strings.TrimSuffix(path, "/")
+}
+
+func (o *RepoOwners) walkFunc(path string, info os.FileInfo, err error) error {
+	log := o.log.WithField("path", path)
+	if err != nil {
+		log.WithError(err).Error("Error while walking OWNERS files.")
+		return nil
+	}
+	filename := filepath.Base(path)
+	if info.Mode().IsDir() {
+		switch filename {
+		case ".git":
+			return filepath.SkipDir
+		case "_output":
+			return filepath.SkipDir
+		}
+	}
+	if !info.Mode().IsRegular() {
+		return nil
+	}
+
+	c := &ownersConfig{}
+	// '.md' files may contain assignees at the top of the file in a yaml header
+	// Note that these assignees only apply to the file itself.
+	if o.enableMDYAML && filename != ownersFileName && strings.HasSuffix(filename, "md") {
+		// Parse the yaml header from the file if it exists and marshal into the config
+		if err := decodeOwnersMdConfig(path, c); err != nil {
+			log.WithError(err).Error("Error decoding OWNERS config from '*.md' file.")
+			return nil
+		}
+
+		// Set assignees for this file (not the directory) using the relative path if they were found
+		relPath, err := filepath.Rel(o.baseDir, path)
+		if err != nil {
+			log.WithError(err).Errorf("Unable to find relative path between baseDir: %q and path.", o.baseDir)
+			return err
+		}
+		o.applyConfigToPath(relPath, c)
+		return nil
+	}
+
+	if filename != ownersFileName {
+		return nil
+	}
+
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to read the OWNERS file.")
+		return nil
+	}
+
+	if err := yaml.Unmarshal(b, c); err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal file contents.")
+		return nil
+	}
+
+	relPath, err := filepath.Rel(o.baseDir, path)
+	if err != nil {
+		log.WithError(err).Errorf("Unable to find relative path between baseDir: %q and path.", o.baseDir)
+		return err
+	}
+	relPathDir := canonicalize(filepath.Dir(relPath))
+	o.applyConfigToPath(relPathDir, c)
+	return nil
+}
+
+// decodeOwnersMdConfig will parse the yaml header if it exists and unmarshal it into an ownersConfig.
```

Comment:
s/decodeOwnersMdConfig/mdStructuredHeaderRegex/

Reply:
That was comment was supposed to be on the function not the regexp. Fixed.